### ISSUE TITLE
Populating "select" field's "option" tags values programmatically

### ISSUE THREAD

**Issue by eatpastacodefasta** [elementor/elementor#10671]
Hi, I'm trying to programmatically fill out the options of a select field with some values.

```
add_filter('elementor_pro/forms/render/item/select', function( $item, $index, $form ) {
	if ( 'my-form' === $form->get_settings_for_display( 'form_name' ) ) {
		if ( '992e235' === $item['_id'] ) {
			$item['options'] = "Red|red\n
			Green|green\n
			Blue|blue";
			$form->add_render_attribute( 'select' . $index, 'value', $item['options'] );
		}
	}
	return $item;
}, 10, 3 );
```
This is all of my code, but it seems that the `add_render_attribute` is not working for the select field. Has anyone managed to make it possible?

-- chris

**Comment by eatpastacodefasta** [comment]
Also tried:

`$form->add_render_attribute( 'option' . $index, 'value', $item['options'] );`
`$form->add_render_attribute( 'select' . $index, 'option', $item['options'] );`

But still nothing 🤔

**Comment by jezremy** [comment]
If you look at the $item var, you will see the $item['options'] is actually $item['field_options']:
`$item['field_options'] = "Red|red\n
Green|green\n
Blue|blue";`

Also, to set the value, instead of $form->add_render_attribute use:
`$item['field_value']` = 'My value';`

You could use $form->add_render_attribute to set a custom attribute, but you don't need to use it for value

**Comment by eatpastacodefasta** [comment]
> If you look at the $item var, you will see the $item['options'] is actually $item['field_options']:
> `$item['field_options'] = "Red|red\nGreen|green\nBlue|blue";`
> 
> Also, to set the value, instead of $form->add_render_attribute use:
> `$item['field_value'] = 'My value';`
> 
> You could use $form->add_render_attribute to set a custom attribute, but you don't need to use it for value

Thank you very much, solved!

Working code:

`add_filter('elementor_pro/forms/render/item/select', function( $item, $index, $form ) {
	if ( 'my-form' === $form->get_settings_for_display( 'form_name' ) ) {
		if ( '66af2e0' === $item['_id'] ) {
			$item['field_options'] = "Red|red\nGreen|green\nBlue|blue";
			$item['field_value'] = 'colors';
		}
	}
	return $item;
}, 10, 3 );`

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v3.3.1

Feel free to update

Thanks!

**Comment by gerardreches** [comment]
> This issue has been resolved in Elementor Pro v3.3.1
> 
> Feel free to update
> 
> Thanks!

This update has also solved #15237
